build-walkthrough.md

```diff
@@ -0,0 +1,171 @@
+**Caveat:** I make not promises that all of this is exactly correct. I think it's _mostly_ correct, but there are almost certainly some omissions and errors.
+
+The key config files:
+
+- `build.gradle`
+- `client/build.gradle`
+- `client.package.json`
+- `client/web pack.*.js`
+- `server/build.gradle`
+
+Let's walk through an trimmed version of the output of running `./gradlew build` on the project.
+
+The main thing the `gradlew` script does is download `gradle` for us so we have it even if it wasn't previously installed on the computer we're sitting in front of. It then passes on any arguments (like `build` in this example) to `gradle`.
+
+After `gradle` is downloaded (if necessary), then `gradle` performs the specified tasks (in this case `build`). To see the list of sub-tasks `gradle` will execute for a given task, use the `-m` flag; if we do `./gradlew -m build` we get:
+
+```
+:server:compileJava SKIPPED
+:server:processResources SKIPPED
+:server:classes SKIPPED
+:server:jar SKIPPED
+:server:startScripts SKIPPED
+:server:distTar SKIPPED
+:server:distZip SKIPPED
+:server:assemble SKIPPED
+:client:nodeSetup SKIPPED
+:client:yarnSetup SKIPPED
+:client:yarn SKIPPED
+:client:yarn_run_test SKIPPED
+:client:runClientTests SKIPPED
+:server:compileTestJava SKIPPED
+:server:processTestResources SKIPPED
+:server:testClasses SKIPPED
+:server:test SKIPPED
+:server:check SKIPPED
+:server:build SKIPPED
+```
+
+These appear in the order in which they will need to be executed, which is sort of the reverse of the dependency graph. If a `gradle` task (e.g., `build`) depends on another `gradle` task (e.g., `check`), then the dependent task (`check`) appears first because it has to be completed before `build` can be considered done. The exact order you get might vary, especially if you (or we) have made some changes to the configuration or the codebase since this was written. Since the dependency graph is really a _graph_ there can be sets of tasks where there is no specified or required ordering among them. At the top level, for example, this has some of the `:server` tasks before the `:client` tasks, but I think that could have done all or most of the `:server` tasks after doing all the `:client` tasks.
```

Comment:
Is `check` really the "dependent task" here? This is a bit nitpick-ey, but wouldn't that imply that `check` depends on `build`, and not the other way around?